routes/items.js

```diff
@@ -2,14 +2,18 @@ const { Router } = require("express");
 const router = Router();
 
 const itemData = require('../dataInterface/items');
+const currentItem = itemData.getById;
 
 router.get("/", (req, res, next) => {
   res.status(200).send(itemData.getAll())
 });
 
 router.get("/:id", (req, res, next) => {
-  // TODO: complete writing this route handler
-  res.status(501).send({ error: 'route not yet implemented' });
+  if (currentItem(req.params.id)) {
+    res.json(itemData.getById(req.params.id))
+  }
+  else (!currentItem(req.params.id))
+  res.status(404).send({ msg: `No item with the id of ${req.params.id}.` })
 });
```

Comment:
In this handler, at least 2 calls to itemData.getById will be made. Each call takes a little time, especially once that data interface is connected to a real database. This would be more efficient of we saved the return value of that itemData.getById call as a variable, so we could check/use its value repeatedly. Something like: 
```
   const foundItem = currentItem(req.params.id)
  if (foundItem) {
    res.status(200).json(foundItem)
  }
  else {
    res.status(404).send({ msg: `No item with the id of ${req.params.id}.` })
  }
```
It's also good practice to specify a 200 status in case of success.

Reply:
Also note that we don't need to get the item again in the else block, as the else block will only execute in the case that the item doesn't exist, as tested in the `if` conditional.